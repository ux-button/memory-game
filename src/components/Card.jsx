function Card ({ id, pokemonName, imageUrl, handleClick }) {
    return (
        <div key={id} className='card' onClick={() => handleClick(id)}>
            <div>
                <img src={imageUrl}></img>
            </div>
            <p>{ pokemonName }</p>
        </div>
    )
}

export { Card }