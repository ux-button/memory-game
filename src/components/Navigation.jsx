import '../App.css'

function NavigationBar ({ score }) {
    return (
        <div className='navigation-bar'>
            Score: {score.points} âœ¦ Best score: {score.best}
        </div>
    )
}

export { NavigationBar }