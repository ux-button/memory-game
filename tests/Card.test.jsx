import {render, screen } from '@testing-library/react'
import { describe, expect } from 'vitest'
import { Card } from '../src/components/Card'

describe('test card component', () => {
    it('match snapshot', () => {
        const handleClick = vi.fn()
        const { container } = render(<Card />);
        expect(container).toMatchFileSnapshot();
    })
})