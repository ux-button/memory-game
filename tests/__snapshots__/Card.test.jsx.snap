// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test card component > match snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <div>
      <img
        height="96px"
        width="96px"
      />
    </div>
    <p />
  </div>
</div>
`;
